frontend/javascripts/oxalis/view/action-bar/volume_actions_view.js

```diff
@@ -1,84 +1,280 @@
 // @flow
-import { Button, Radio, Tooltip } from "antd";
-import { connect } from "react-redux";
-import React, { PureComponent } from "react";
+import { Button, Radio, Tooltip, Badge } from "antd";
+import { useSelector } from "react-redux";
+import React, { useEffect } from "react";
+import { usePrevious, useKeyPress } from "libs/react_hooks";
 
-import { type VolumeTool, VolumeToolEnum } from "oxalis/constants";
+import {
+  ToolsWithOverwriteCapabilities,
+  type VolumeTool,
+  VolumeToolEnum,
+  type OverwriteMode,
+  OverwriteModeEnum,
+} from "oxalis/constants";
 import { document } from "libs/window";
 import { enforceVolumeTracing } from "oxalis/model/accessors/volumetracing_accessor";
-import { setToolAction, createCellAction } from "oxalis/model/actions/volumetracing_actions";
+import {
+  setToolAction,
+  createCellAction,
+  setOverwriteModeAction,
+} from "oxalis/model/actions/volumetracing_actions";
 import ButtonComponent from "oxalis/view/components/button_component";
-import Store, { type OxalisState } from "oxalis/store";
+import Store from "oxalis/store";
+import { convertCellIdToCSS } from "oxalis/view/right-menu/mapping_info_view";
 
-// Workaround until github.com/facebook/flow/issues/1113 is fixed
-const RadioGroup = Radio.Group;
-const RadioButton = Radio.Button;
-const ButtonGroup = Button.Group;
+function getMoveToolHint(activeTool, isShiftPressed, isControlPressed, isAltPressed): ?string {
+  if (activeTool !== VolumeToolEnum.MOVE) {
+    return null;
+  }
 
-type Props = {|
-  activeTool: VolumeTool,
-  isInMergerMode: boolean,
-|};
+  if (!isShiftPressed && !isControlPressed && !isAltPressed) {
+    return null;
+  }
 
-class VolumeActionsView extends PureComponent<Props> {
-  componentDidUpdate = (prevProps: Props) => {
-    if (!prevProps.isInMergerMode && this.props.isInMergerMode) {
-      Store.dispatch(setToolAction(VolumeToolEnum.MOVE));
+  if (isShiftPressed && !isControlPressed && !isAltPressed) {
+    return "Click to select a node.";
+  }
+
+  if (!isShiftPressed && isControlPressed && !isAltPressed) {
+    return "Drag to move the selected node. Right-click to create a new node without selecting it.";
+  }
+
+  if (isShiftPressed && !isControlPressed && isAltPressed) {
+    return "Click on a node to create an edge to the currently active node.";
+  }
+
+  if (isShiftPressed && isControlPressed && !isAltPressed) {
+    return "Click on a node to delete the edge to the currently active node.";
+  }
+
```

Comment:
I think, this is the if-case a few lines up:

```
  if (isShiftPressed && !isControlPressed && isAltPressed) {
    return "Click on a node to create an edge to the currently active node.";
  }
```

It doesn't mention that two trees can be merged, though. Do you think we should add this? However, I want to avoid long tooltips as they can look a bit weird :see_no_evil:

Reply:
I see, two things:
If shift is hit first, the tooltip doesn't change to the "Click on a node to create..." but remains "Click to select a node" - That's why I thought this particular combination was missing.
Also, as this case can only be used to merge two different trees (not to create an edge to a node in the same tree) I would mention it in the tooltip. Maybe "Click on a node in another tree to merge the two trees." or with an added " by adding an edge." Hmm, not sure but it can always be refined, feel free to also leave as-is.

Reply:
Ah, disregard my earlier comment, I didn't see yours here. Will work on it now!

Reply:
> If shift is hit first, the tooltip doesn't change to the "Click on a node to create..." but remains "Click to select a node" - That's why I thought this particular combination was missing.

Glad you noticed this! I could reproduce and also fix it finally. It was a weird bug, though. When pressing shift and alt and then releasing alt, the `key` property contained `Meta` even though that key wasn't even touched. I worked around this by explicitly inspecting the `altKey` property in the key-up/down events. Works now fine for me!

> Also, as this case can only be used to merge two different trees (not to create an edge to a node in the same tree) I would mention it in the tooltip. Maybe "Click on a node in another tree to merge the two trees." or with an added " by adding an edge." Hmm, not sure but it can always be refined, feel free to also leave as-is.

Very good point! I forgot that one can only merge nodes if they are from different trees. Since Alt+Drag moves the clicked window on my linux setup, I didn't really play around much with this :/ I adapted the message according to your suggestion :100:

Reply:
Glad you found a fix for this weird behavior (I found this thread, seems like this is by design: https://askubuntu.com/questions/567731/why-is-shift-alt-being-mapped-to-meta :see_no_evil:)

Reply:
Ah, cool, that you found some resources on this. I guess, we could also tell all our linux users they should run `xmodmap -e "keycode 64 = Alt_L"` :rofl: